applications/StructuralMechanicsApplication/python_scripts/structural_response_function_factory.py

```diff
@@ -25,6 +25,9 @@ def CreateResponseFunction(response_id, response_settings, model):
     elif response_type == "adjoint_local_stress":
         return structural_response.AdjointResponseFunction(response_id, response_settings, model)
 
+    elif response_type == "global_finite_differencing":
```

Comment:
This breaks a bit the logic of this factory, as the additional response function just refers to one part of what I would expect to be a response, meaning the sensitivity part. 

While with the other options one really specifies the type of response (possibly along with a particular sensitivity analysis, i.e. adjoint analysis), the option for global FD only specifies the gradient analysis and then internally the actual response type. This calls for the "sub_response_settings" you introduced. However, this has some more implications that I detailed a bit further down in a comment.